### PR TITLE
Update enterprise packaging in feature docs

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
The following changes were made:
- Each Enterprise docs page makes it clear which Consul Enterprise packages include that feature
- The Enterprise overview page
  - provides a table summarizing the features available per package
  - fixes/improves language regarding the license
  
### Preview
- [Enterprise overview page](https://consul-4dfr5hup6-hashicorp.vercel.app/docs/enterprise)
- [Enterprise features matrix](https://consul-4dfr5hup6-hashicorp.vercel.app/docs/enterprise#consul-enterprise-feature-availability)
- [Audit logging](https://consul-4dfr5hup6-hashicorp.vercel.app/docs/enterprise/audit-logging) - as one example of an individual docs page that was updated

### Questions
Do we agree with the re-organization of the enterprise features page to make the "types" of features more clear?

image

Post-merge work:
- [ ] Make similar changes to Consul guides on learn.hashicorp.com

### CODE REVIEW

File: website/content/docs/enterprise/audit-logging.mdx [hunk 1]
```diff
@@ -8,9 +8,11 @@ description: >-
 # Audit Logging
 
 <EnterpriseAlert>
-  This feature requires{' '}
-  <a href="https://www.hashicorp.com/products/consul/">Consul Enterprise</a>{' '}
-  with the Governance and Policy module.
+  This feature is available in v1.8.0+ of: (1) {' '}
```
Comment (jkirschner-hashicorp) [inline website/content/docs/enterprise/audit-logging.mdx:11]: **Question for reviewer**

Do we agree with including the version inline here? (versus just `is available in: (1) ...`

And should it be `v1.8.0+` or `versions 1.8.0+`?

File: website/content/docs/enterprise/audit-logging.mdx [hunk 2]
```diff
@@ -8,9 +8,11 @@ description: >-
...
+  <a href="https://cloud.hashicorp.com/pricing/consul/">HashiCorp Cloud Platform (HCP) Consul - Standard tier and above</a>
```
Comment (jkirschner-hashicorp) [inline website/content/docs/enterprise/audit-logging.mdx:12]: @pcmccarron : is this the link we want to use for the link circled in red below? Or some other HCP link?

image

File: website/content/docs/enterprise/index.mdx [hunk 3]
```diff
@@ -8,21 +8,31 @@ description: >-
...
-- [Redundancy Zones](/docs/enterprise/redundancy)
-- [Advanced Federation for Complex Network Topologies](/docs/enterprise/federation)
-- [Network Segments](/docs/enterprise/network-segments)
-- [Namespaces](/docs/enterprise/namespaces)
-- [NIA with Terraform Enterprise](/docs/nia/enterprise)
-- [Sentinel](/docs/enterprise/sentinel)
-- [OIDC Auth Method](/docs/acl/auth-methods/oidc)
+| Feature                                                                           | [HashiCorp Cloud Platform (HCP) Consul]                   | [Consul Enterprise] | Legacy Consul Enterprise (module-based)           |
+| --------------------------------------------------------------------------------- | ------------------------------------------------------- | ----------------- | ------------------------------------------------- |
+| Consul servers as a managed service                                               | Yes                                                     | No (self-managed) | No (self-managed)                                 |
```
Comment (jkirschner-hashicorp) [inline website/content/docs/enterprise/index.mdx:24]: @pcmccarron : Do you want to keep this table row (in red below)? Anything you'd change about it?

image

File: website/content/docs/enterprise/audit-logging.mdx (hunk 2 above)
Comment (trujillo-adam) [reply website/content/docs/enterprise/audit-logging.mdx:12]: Per the comment above, we should probably just link to the matrix on the CE

File: website/content/docs/enterprise/index.mdx [hunk 4]
```diff
@@ -8,21 +8,31 @@ description: >-
...
+| [Audit Logging](/docs/enterprise/audit-logging)                                   | Standard tier and above                                 | Yes               | With Governance and Policy module                 |
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/index.mdx:25]: My only suggestion is to use regular sentence-style capitalization for each cell, instead of capitalizing each feature/module

**Review by trujillo-adam — COMMENTED** [review]
The comment the admin partitions applies to all sections.

File: website/content/docs/enterprise/audit-logging.mdx (hunk 2 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/audit-logging.mdx:12]: @trujillo-adam: anything else you'd change about this?

image

File: website/content/docs/enterprise/index.mdx (hunk 4 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/index.mdx]: I missed this change in my last pass. I'll address after the next review (along with any other new feedback).

File: website/content/docs/enterprise/federation.mdx [hunk 5]
```diff
@@ -10,9 +10,9 @@ description: >-
 # Consul Enterprise Advanced Federation
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/federation.mdx:11]: This is kind of a pointless page, regardless of the link to the matrix. Users are going to be frustrated that there isn't information about how to use the feature. We don't even link to the applicable information in the docs (which I believe is the [Service-to-service Traffic Across Partitions](https://www.consul.io/docs/connect/gateways/mesh-gateway/service-to-service-traffic-partitions#service-to-service-traffic-across-partitions) page). My recommendation is to work the appropriate content into the core docs and link from the core docs to the matrix.

File: website/content/docs/enterprise/index.mdx [hunk 6]
```diff
@@ -8,22 +8,34 @@ description: >-
 
 # Consul Enterprise
 
-[Consul Enterprise](https://www.hashicorp.com/consul) features address the organizational complexities of collaboration, operations, scale and governance. It includes capabilities that improve failure resilience, read scalability, and managing access in accordance with organizational policies. It also adds support for application and service architectures across complex network topologies. If you have already purchased Consul Enterprise, please see the [licensing section](#licensing) below.
+[Consul Enterprise](https://www.hashicorp.com/products/consul/)
+features address the organizational complexities of collaboration, operations, scale, and governance
+and are [available in several forms of Consul Enterprise](/docs/enterprise/feature-availability).
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/index.mdx:13]:
```suggestion
features address the organizational complexities of collaboration, operations, scale, and governance
and are available in several forms of Consul Enterprise. Refer to the [feature availability matrix](/docs/enterprise/feature-availability) for details.
```
It's a (slightly) better user experience to give an idea where you are sending readers.

File: website/content/docs/enterprise/audit-logging.mdx [hunk 7]
```diff
@@ -8,9 +8,9 @@ description: >-
 # Audit Logging
 
 <EnterpriseAlert>
-  This feature requires{' '}
-  <a href="https://www.hashicorp.com/products/consul/">Consul Enterprise</a>{' '}
-  with the Governance and Policy module.
+  This feature is available in versions 1.8.0+ of
+  HashiCorp Cloud Platform (HCP) and self-managed Consul according to the{' '}
+  <a href="/docs/enterprise/feature-availability">enterprise feature matrix</a>.
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/audit-logging.mdx:13]:
```suggestion
  This feature is requires version 1.8.0+ of
  HashiCorp Cloud Platform (HCP) or self-managed Consul. Refer to the {' '}
  <a href="/docs/enterprise/feature-availability">enterprise feature matrix</a> for additional information.
```
Just to reiterate, it would be better to specify in a 'Requirements' section.  
We can do that down the road. For now, I suggest rewording all of these instances to fix the odd cause/effect construction " . . . according to the enterprise feature matrix".

**Review by trujillo-adam — CHANGES_REQUESTED** [review]
At a minimum, I think we should merge the FA content into the overview page.

File: website/content/docs/enterprise/admin-partitions.mdx [hunk 8]
```diff
@@ -7,9 +7,9 @@ description: Consul Enterprise enables you to create partitions that can be admi
 # Consul Enterprise Admin Partitions
 
 <EnterpriseAlert>
-  This feature requires{' '}
-  <a href="https://www.hashicorp.com/products/consul/">Consul Enterprise</a>{' '}
-  with the Governance and Policy module.
+  This feature is available in versions 1.11.0+ of
+  HashiCorp Cloud Platform (HCP) and self-managed Consul according to the{' '}
+  <a href="/docs/enterprise/feature-availability">enterprise feature matrix</a>.
```
[the remark this answers is not in the document]
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/admin-partitions.mdx:12]: Ah, meaning you're recommending moving this content to a Requirements section and removing the "notice box" from the top? I suppose that does make more sense ... people probably want to know what the thing is first and whether it's valuable / useful to them, not what version they need :)

Do you think it would still be useful to give them a link here to the matrix of all features for Consul Enterprise (and its different forms)?

File: website/content/docs/enterprise/feature-availability.mdx [hunk 9]
```diff
@@ -0,0 +1,47 @@
+---
```
[the remark this answers is not in the document]
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/feature-availability.mdx:1]: I agree, but there was internal disagreement on this topic. Will open a Slack thread to discuss.

File: website/content/docs/enterprise/federation.mdx (hunk 5 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/federation.mdx:11]: I agree, though I think making this page more useful is out of scope for this PR.

This reminds me of how the network segments docs page [used to be](https://www.consul.io/docs/v1.9.x/enterprise/network-segments) versus [how it is now](https://www.consul.io/docs/enterprise/network-segments). I think we can make it more useful going forward (like we did for the network segments page).

I do have other concerns about this page. For example, it's ostensibly about the network areas feature, but network areas isn't the title of the page or how it's described in the feature list. This page doesn't come up at all if you search "network areas" on the site (though maybe that's actually OK for now because the current state of the page is not that helpful!)

File: website/content/docs/enterprise/feature-availability.mdx (hunk 9 above)
[the remark this answers is not in the document]
Comment (karl-cardenas-coding) [reply website/content/docs/enterprise/feature-availability.mdx:1]: Let's go with the overview page.

File: website/content/docs/enterprise/audit-logging.mdx (hunk 7 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/audit-logging.mdx]: Reworded accordingly. Moving to a requirements section can happen in a future PR.

File: website/content/docs/enterprise/index.mdx (hunk 6 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/index.mdx]: Done

**Comment by jkirschner-hashicorp** [comment]
@karl-cardenas-coding , @trujillo-adam : I believe this is ready for final review. I moved the "feature availability" back to the main overview page. Here are links to some previews:

- [Enterprise overview page](https://consul-qpompr25b-hashicorp.vercel.app/docs/enterprise)
- [Enterprise features matrix](https://consul-qpompr25b-hashicorp.vercel.app/docs/enterprise#consul-enterprise-feature-availability)
- [Audit logging](https://consul-qpompr25b-hashicorp.vercel.app/docs/enterprise/audit-logging) - as one example of an individual docs page that was updated

File: website/content/docs/enterprise/index.mdx [hunk 10]
```diff
@@ -8,22 +8,73 @@ description: >-
...
+### Self-Managed Consul
+
+To gain access to Consul Enterprise features in a self-managed installation,
+[apply a purchased license](/docs/enterprise/license/overview)
+to the Consul Enterprise binary that grants access to the desired features.
+
+### HCP Consul
+
+No action is required to gain access to Consul Enterprise features in a
+HashiCorp Cloud Platform installation.
```
Comment (karl-cardenas-coding) [inline website/content/docs/enterprise/index.mdx:54]:
```suggestion
[HashiCorp Cloud Platform](https://cloud.hashicorp.com/products/consul) installation.
```

Adding link to HCP.

File: website/content/docs/enterprise/index.mdx [hunk 11]
```diff
@@ -8,22 +8,73 @@ description: >-
...
+- [Sentinel for KV](/docs/enterprise/sentinel): policy-as-code framework for defining advanced key-value storage access control policies
+
+# Gain Access to Enterprise Features
+
+### Self-Managed Consul
+
+To gain access to Consul Enterprise features in a self-managed installation,
+[apply a purchased license](/docs/enterprise/license/overview)
+to the Consul Enterprise binary that grants access to the desired features.
+
```
Comment (karl-cardenas-coding) [inline website/content/docs/enterprise/index.mdx:50]:
```suggestion
To gain access to Consul Enterprise features in a self-managed installation,
[apply a purchased license](/docs/enterprise/license/overview)
to the Consul Enterprise binary that grants access to the desired features.
You can also try out Consul Enterprise through a 30 day trial license. Visit [https://consul.io/trial](https://consul.io/tria) for a trial license. 

```
I think this will help guide users towards the trial page :)

File: website/content/docs/enterprise/index.mdx [hunk 12]
```diff
@@ -8,22 +8,73 @@ description: >-
...
+Choose the form that best suits the needs of your organization.
+
+Available enterprise features per Consul form and license include:
+
+| Feature                                                  | [HashiCorp Cloud Platform (HCP) Consul]                 | [Consul Enterprise] | Legacy Consul Enterprise (module-based)           |
+| -------------------------------------------------------- | ------------------------------------------------------- | ------------------- | ------------------------------------------------- |
+| Consul servers as a managed service                      | Yes                                                     | No (self-managed)   | No (self-managed)                                 |
+| [Admin Partitions](/docs/enterprise/admin-partitions)    | Standard tier and above                                 | Yes                 | With Governance and Policy module                 |
+| [Namespaces](/docs/enterprise/namespaces)                | Standard tier and above                                 | Yes                 | With Governance and Policy module                 |
```
Comment (karl-cardenas-coding) [inline website/content/docs/enterprise/index.mdx:67]: @jkirschner-hashicorp I like the table. The one piece I am not a fan of are the empty slots under HCP. I understand that some of these features are not applicable to HCP but we should indicate as such. This will prevent customers asking questions about these features in HCP.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Looks great @jkirschner-hashicorp. I provided some minor suggestions, and left you a question for the table.

File: website/content/docs/enterprise/index.mdx (hunk 12 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/index.mdx:67]: Thanks @karl-cardenas-coding. Just synced with @srikrishmurthy on this. I'll make a few changes (one of these features might be truly "Not Applicable" in the case of HCP Consul).

File: website/content/docs/enterprise/index.mdx (hunk 11 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/enterprise/index.mdx:50]: I'm also going to add a similar call-out to the HCP section.

**Comment by jkirschner-hashicorp** [comment]
@karl-cardenas-coding : made changes based on your feedback and conversation with @srikrishmurthy . New previews:
- [Enterprise overview page](https://consul-4dfr5hup6-hashicorp.vercel.app/docs/enterprise)
- [Enterprise features matrix](https://consul-4dfr5hup6-hashicorp.vercel.app/docs/enterprise#consul-enterprise-feature-availability)
- [Audit logging](https://consul-4dfr5hup6-hashicorp.vercel.app/docs/enterprise/audit-logging) - as one example of an individual docs page that was updated

**Comment by karl-cardenas-coding** [comment]
@jkirschner-hashicorp I still see empty cells
image

**Comment by jkirschner-hashicorp** [comment]
@karl-cardenas-coding : I added "No" to each of the empty cells. Ready for final review!

**Review by karl-cardenas-coding — APPROVED** [review]
👍🏼

File: website/content/docs/enterprise/index.mdx [hunk 13]
```diff
@@ -8,22 +8,79 @@ description: >-
...
-
-These features are part of [Consul
-Enterprise](https://www.hashicorp.com/consul).
+[Consul Enterprise](https://www.hashicorp.com/products/consul/)
+features address the organizational complexities of collaboration, operations, scale, and governance
+and are [available in several forms of Consul Enterprise](#consul-enterprise-feature-availability).
+If you have purchased Consul Enterprise, refer to
+[how to gain access to enterprise features](#gain-access-to-enterprise-features).
+
+Enterprise features include:
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/index.mdx:17]:
```suggestion
If you have purchased a Consul Enterprise license, refer to
[Access Your Enterprise Features](#access-your-enterprise-features).

Enterprise features include:
```
"Gaining" the feature features sounds weird -- like gaining experience points or something

File: website/content/docs/enterprise/index.mdx [hunk 14]
```diff
@@ -8,22 +8,79 @@ description: >-
...
+
+#### Multi-Tenancy
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/index.mdx:19]: Did we have a discussion about properly nesting headings? These should be ## . I still think it should be "Multi-tenancy" because compound words are considered a single word, but we're not consistent enough on this to enforce.

File: website/content/docs/enterprise/index.mdx [hunk 15]
```diff
@@ -8,22 +8,79 @@ description: >-
...
+- [Admin Partitions](/docs/enterprise/admin-partitions): define administrative boundaries between tenants within a single Consul datacenter
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/index.mdx:20]:
```suggestion
- [Admin Partitions](/docs/enterprise/admin-partitions): Define administrative boundaries between tenants within a single Consul datacenter.
```
We should use complete sentence with appropriate caps and punctuation, even if there are a few sentence fragments on this page.

File: website/content/docs/enterprise/index.mdx [hunk 16]
```diff
@@ -8,22 +8,79 @@ description: >-
...
+#### Complex Network Topology Support
+- [Network Areas](/docs/enterprise/federation): support complex network topologies between federated Consul datacenters with pairwise federation rather than full mesh federation
+- [Network Segments](/docs/enterprise/network-segments): support complex network topologies within a Consul datacenter by enforcing boundaries in Consul client gossip traffic
+
+#### Governance
+- [OIDC Auth Method](/docs/security/acl/auth-methods/oidc): manage user access to Consul through an OIDC identity provider instead of Consul ACL tokens directly
+- [Audit Logging](/docs/enterprise/audit-logging): understand Consul access and usage patterns by reviewing access to the Consul HTTP API 
+- [Sentinel for KV](/docs/enterprise/sentinel): policy-as-code framework for defining advanced key-value storage access control policies
+
+# Gain Access to Enterprise Features
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/index.mdx:43]:
```suggestion
## Access Your Enterprise Features

The following sections describe how to access enterprise features for each Consul distribution.
```
There should only be one # H1 per page

File: website/content/docs/enterprise/index.mdx [hunk 17]
```diff
@@ -8,22 +8,79 @@ description: >-
...
+
+### HCP Consul
+
+No action is required to gain access to Consul Enterprise features in a
+[HashiCorp Cloud Platform](https://cloud.hashicorp.com/products/consul) installation.
+
+You can try out HCP Consul for free. Refer to the
+[HCP Consul product page](https://cloud.hashicorp.com/products/consul) for more details.
+
+# Consul Enterprise Feature Availability
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/index.mdx:62]:
```suggestion
## Consul Enterprise Feature Availability
```

File: website/content/docs/enterprise/namespaces.mdx [hunk 18]
```diff
@@ -7,11 +7,14 @@ description: Consul Enterprise enables data isolation with Namespaces.
...
-  <a href="https://www.hashicorp.com/products/consul/">Consul Enterprise</a>{' '}.
+  This feature requires version 1.7.0+ of
+  HashiCorp Cloud Platform (HCP) or self-managed Consul Enterprise.
+  Refer to the{' '}
+  <a href="/docs/enterprise#consul-enterprise-feature-availability">enterprise feature matrix</a>
+  {' '}for additional information.
 </EnterpriseAlert>
 
-With Consul Enterprise v1.7.0, data for different users or teams
+With Consul Enterprise 1.7.0+, data for different users or teams
```
Comment (trujillo-adam) [inline website/content/docs/enterprise/namespaces.mdx:17]:
```suggestion
With Consul Enterprise 1.7.0+, data for different users or teams
```
Do we need to repeat the version-specific information here? It's in the EnterpriseAlert box. I don't see that we're doing it in the following pages.

**Review by trujillo-adam — CHANGES_REQUESTED** [review]
Still have a few things that we should figure out

**Review by trujillo-adam — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/655909.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/655909)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `release/1.12.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/655909)

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/655928.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/655928)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `release/1.12.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/655928)

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/655942.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/655942)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `release/1.12.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/655942)

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/656862.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `release/1.12.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/656882.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/656882)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit d04fe6ca2c1014d11cfebea9069abf90653fdc22 onto `release/1.12.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/656882)

**Comment by jkirschner-hashicorp** [comment]
This has now been successfully deployed to the website (cherry-picked onto `stable-website` and `release/1.12.x`).
